packages/e2e-tests/specs/rich-text.test.js

```diff
@@ -214,4 +214,30 @@ describe( 'RichText', () => {
 
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	it( 'should update internal selection after fresh focus', async () => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '1' );
+		// Focus the address bar (move focus out of the window).
+		await pressKeyWithModifier( 'primary', 'l' );
+		// Ensure the element lost focus.
+		// Stangely enough, this is not needed when testing manually.
+		await page.evaluate( () => {
+			document.querySelector( '.wp-block-paragraph' ).blur();
+		} );
+		// Focus the rich text instance again (not at the start).
+		await page.click( '.wp-block-paragraph', { button: 'middle' } );
+		// Wait for animation frame.
+		await page.evaluate( async () => {
+			await new Promise( ( resolve ) => {
+				window.requestAnimationFrame( resolve );
```

Comment:
This is a bit of a problem... Ideally we should have updated the internal selection before Puppeteer resolves the `click` promise. I could try to fix this, but we risk optimising for Puppeteer and Chrome. I'll have to check other browsers if the order of events is the same and if the fix works there as well. Cc @gziolo as a fix could potentially resolve intermittent failures.